neuralprophet/configure.py

```diff
@@ -161,19 +171,41 @@ def set_auto_batch_epoch(
         # also set lambda_delay:
         self.lambda_delay = int(self.reg_delay_pct * self.epochs)
 
-    def get_optimizer(self, model_parameters):
-        return utils_torch.create_optimizer_from_config(self.optimizer, model_parameters, self.learning_rate)
-
-    def get_scheduler(self, optimizer, steps_per_epoch):
-        return torch.optim.lr_scheduler.OneCycleLR(
-            optimizer,
-            max_lr=self.learning_rate,
-            epochs=self.epochs,
-            steps_per_epoch=steps_per_epoch,
-            pct_start=0.3,
-            anneal_strategy="cos",
-            div_factor=100.0,
-            final_div_factor=5000.0,
+    def set_optimizer(self):
+        """
+        Set the optimizer and optimizer args. If optimizer is a string, then it will be converted to the corresponding torch optimizer.
+        The optimizer is not initialized yet as this is done in configure_optimizers in TimeNet.
+        """
+        self.optimizer, self.optimizer_args = utils_torch.create_optimizer_from_config(
+            self.optimizer, self.optimizer_args
+        )
```

Comment:
@ourownstory I would propose to move the logic from `create_optimizer_from_config` directly into the `set_optimzer` function since that's how we also handle it for `set_loss_func` and the other setter functions. What do you think?